modules/core/04-channel/v2/keeper/msg_server_test.go

```diff
@@ -337,3 +337,119 @@ func (suite *KeeperTestSuite) TestProvideCounterparty() {
 		}
 	}
 }
+
+func (suite *KeeperTestSuite) TestMsgAcknowledgement() {
+	var (
+		path         *ibctesting.Path
+		msgAckPacket *channeltypesv2.MsgAcknowledgement
+		recvPacket   channeltypesv2.Packet
+	)
+	testCases := []struct {
+		name     string
+		malleate func()
+		expError error
+	}{
+		{
+			name:     "success",
+			malleate: func() {},
+		},
+		{
+			name: "success: NoOp",
+			malleate: func() {
+				suite.chainA.App.GetIBCKeeper().ChannelKeeperV2.SetPacketCommitment(suite.chainA.GetContext(), recvPacket.SourceChannel, recvPacket.Sequence, channeltypesv2.CommitPacket(recvPacket))
```

Comment:
might be nice to verify app callback wasn't executed, maybe not executed. We could do that by toggling a bool in the ack app callback, but would require a slight refactor to the test to allow custom assertions post test, will leave it to you if you think it's worth it.

Reply:
Interestingly, this makes the test fail! I think it's either my setpacketcommitment that is wrong, or (maybe) an error in the code, thanks for the idea!

Reply:
Instead of changing the structure of the code, I modified the callback to return an error, so that we can simply check that an error is not returned, WDYT? (the comment explains it better :D)